sdk/core/azure-core/test/ut/logging_test.cpp

```diff
@@ -202,3 +202,94 @@ TEST(Logger, Message)
     throw;
   }
 }
+
+TEST(Logger, LoggerStream)
+{
+  Logger::Level level = Logger::Level::Error;
+  std::string message;
+
+  Logger::SetListener([&](auto lvl, auto msg) {
+    level = lvl;
+    message = msg;
+  });
+
+  Logger::SetLevel(Logger::Level::Verbose);
+  {
+    Log::Stream(Logger::Level::Verbose) << "Verbose" << std::endl;
+    EXPECT_EQ(message, "Verbose\n");
```

Comment:
So, we will be getting log messages with `\n`s? That is not convenient.
And how should we have messages with newlines within a single message?